### PR TITLE
Use yao instead of fog

### PR DESCRIPTION
**Pull request by Joe-noh**
This PR replaces fog with yao.

With this, the configuration file `.fog` is not longer required and `.kaname` will be used instead.

### CODE REVIEW

File: test/test_kaname_adapter_real.rb [hunk 1]
```diff
@@ -2,13 +2,27 @@
...
+        Kaname::Config.stub(:management_url, dummy_management_url) do
+          Kaname::Config.stub(:username, "dummy_username") do
+            Kaname::Adapter::Real.new.update_user_password('old', 'new')
+          end
+        end
+      end
+    end
+
+    mock_auth.verify
+    mock_user.verify
```
Comment (udzura) [inline test/test_kaname_adapter_real.rb:26]: All of existing tests other than this passed? Great!

File: kaname.gemspec [hunk 2]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "fog"
+  spec.add_dependency "yao"
```
Comment (udzura) [inline kaname.gemspec]: It seems that `yao`'s dependency should be set to `~> 0.1.0`

File: lib/kaname/config.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      config_file = File.join(Dir.home, '.kaname')
+      raise '~/.kaname is missing' unless File.exists?(config_file)
+
+      config = YAML.load_file(config_file)
+
+      @@auth_url       = config['auth_url']
+      @@tenant         = config['tenant']
+      @@username       = config['username']
+      @@password       = config['password']
+      @@management_url = config['management_url']
```
Comment (udzura) [inline lib/kaname/config.rb:35]: If some of necessary parameters are empty, I want to get error here.

Comment (Joe-noh) [inline lib/kaname/config.rb:35]: fixed.

File: lib/kaname.rb [hunk 4]
```diff
@@ -7,3 +9,5 @@
 
 module Kaname
 end
+
+Kaname::Config.setup unless ENV['RACK_ENV'] == 'test'
```
Comment (udzura) [inline lib/kaname.rb:13]: minitest will load `minitest_helper.rb` first on running test suite.

https://github.com/yaocloud/kaname/blob/master/test/minitest_helper.rb

`Kaname::Config.setup` can be moved to that location. (And `RACK_ENV` can be removed!)

File: lib/kaname/adapter/real.rb [hunk 5]
```diff
@@ -5,37 +5,43 @@ module Kaname
...
-            http.request(req)
-          }
+
+          res = Net::HTTP.start(url.host, url.port) {|http| http.request(req) }
+
           if res.code == "200"
-            puts "Your password is updated. Please update your ~/.fog configuration too."
+            puts "Your password is updated. Please update your ~/.kaname configuration too."
           else
             raise "password updating is failed"
```
Comment (udzura) [inline lib/kaname/adapter/real.rb:46]: These diff looks :+1: to me

File: Rakefile [hunk 6]
```diff
@@ -2,6 +2,8 @@ require "bundler/gem_tasks"
 require 'rake/testtask'
 
 Rake::TestTask.new do |t|
+  ENV['RACK_ENV'] = 'test'
```
Comment (hsbt) [inline Rakefile]: I prefer to move this line like test specific configuration to `minitest_helper.rb`

**Comment by hsbt** [comment]
@Joe-noh Thank you for your effort. It's a great step for yaocloud.

**Pull request merged**